test_programs/compile_failure/unsigned_integer_literal_underflow/src/main.nr

```diff
@@ -0,0 +1,5 @@
+fn main() {
+    foo(-1)
+}
+
+fn foo(_x: u8) {}
```

Comment:
Since you have a few small tests in this PR which do not rely on the stdlib, I think `noirc_frontend/src/tests.rs` may be a better place for them. There, you can still assert that they error but we can place them in smaller unit tests.

Reply:
Ooooh, this is great! I didn't know such file existed. Tomorrow I'll change this PR to use that approach.

Reply:
I tried to do this but it seems errors from the SSA pass don't get captured here... which maybe makes sense! I'm thinking ideally the error happens before SSA, but that refactor doesn't seem that easy to do (I can't find a place where integer ranges are checked on the HIR side, maybe because that code doesn't exist yet).

I'll merge this PR and later if I have some time I can try to do this refactor.

Reply:
Ah, it's not approved yet. I'll try a bit more then 😊

Reply:
Hm, it seems integer literals are assigned a type at the monomorphization stage... and that also doesn't run in these tests. I was thinking about moving the error from SSA to monomorphization, but it wouldn't trigger either.

I think I'll need to understand a bit more all of the compiler passes, what they do and when they run.

Reply:
(so for now I won't make any more changes to this PR... except maybe improving the error message a bit to show what the valid ranges are, when an error happens)

Reply:
> I tried to do this but it seems errors from the SSA pass don't get captured here.

Ah, right. Forgot this check is done in SSA - my bad. Yes, the frontend tests only run name resolution & type checking. They stop before monomorphization so this won't be useful here. No need to refactor this PR.